pkg/kubelet/kubelet.go

```diff
@@ -2931,8 +2931,15 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontaine
 			return
 		}
 		if kl.isPodResizeInProgress(allocatedPod, podStatus) {
+			// Preserve the old reason and message if there is one, to avoid clearing error messages here.
+			var oc v1.PodCondition
+			for _, c := range kl.statusManager.GetPodResizeConditions(pod.UID) {
+				if c.Type == v1.PodResizeInProgress {
+					oc = *c
+				}
+			}
```

Comment:
I recommend moving this logic into `statusManager.SetPodResizeInProgressCondition`: if the incoming `Reason` is `""` (or reason & message?), then preserve the previous reason & message.

Reply:
I originally didn't put this in `statusManager.SetPodResizeInProgressCondition` because I wanted to be able to clear the error message (without needing to clear the entire condition) when `doPodResizeAction` succeeded. I updated this so that this is now in `statusManager.SetPodResizeInProgressCondition`, but also added a boolean `allowReasonToBeCleared` to toggle this